hack/tools/go.mod

```diff
@@ -49,6 +49,8 @@ require (
 	github.com/blang/semver/v4 v4.0.0 // indirect
 	github.com/cespare/xxhash/v2 v2.1.2 // indirect
 	github.com/containerd/containerd v1.6.12 // indirect
+	github.com/coredns/caddy v1.1.0 // indirect
+	github.com/coredns/corefile-migration v1.0.18 // indirect
```

Comment:
q for my understanding, why are there such changes in hack/tools/go.mod? this sems unrelated to the other changes in this PR

Reply:
I am also not sure why these changes are in the PR. But `go mod tidy` and the Prow job want this change.

IIRC we found out why these changes happens @chrischdi, right? Wo do you still know why?

Reply:
hack/tools/tilt-prepare uses "sigs.k8s.io/cluster-api/cmd/clusterct/*"

"sigs.k8s.io/cluster-api/cmd/clusterct/*" now uses "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1" (for the KubeadmControlPlane type)

"sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1" uses "github.com/coredns/corefile-migration" (for the KCP webhook)

"github.com/coredns/corefile-migration" uses "github.com/coredns/caddy" (internally)

Reply:
I came to the same conclusion as stefan 🏆